### ISSUE TITLE
return 'enum' value

### ISSUE THREAD

**Issue by nico-robert** [apnadkarni/tcl-cffi#61]
Hello @apnadkarni ,
I think it's not possible to return a enum value... But it's possible to return enum by reference ?
```
cffi::enum define EncoderType {
    SINGLE_BYTE 0
    DOUBLE_BYTE 1
    UNINITIALIZED 2
    UNKNOWN 3
}
```
and my function :
`LIB stdcall GetEncoder int {h pointer.int encoder {enum.EncoderType out}}`

I have this error : _Invalid value "enum.EncoderType". Invalid base type._

Do you have a workaround ?

Thanks
Nicolas

**Comment by apnadkarni** [comment]
Yes, right now enums cannot be used as return values or output types. Only workaround right now is to just use plain old ints.

There was some reason or ambiguity why it was not added for output parameters that I don't recall but certainly it makes sense and it is non-intuitive to only have it for input.

I'll look at fixing for the next release.

As an aside, the syntax will be a little different from what you have shown for output enums.

/Ashok

**Comment by nico-robert** [comment]
ok , I understand , thanks

**Comment by apnadkarni** [comment]
Re-opening as feature request.

**Comment by apnadkarni** [comment]
Implemented for 1.0b8
